src/FunctionalTest/Smo/ScriptingTests/PartitionFunction_SmoTestSuite.cs

```diff
@@ -60,6 +62,32 @@ public void SmoDropIfExists_PartitionFunction_Sql16AndAfterOnPrem()
                 });
         }
 
+        /// <summary>
+        /// Tests create script of a partition function handling DBNull range values correctly
+        /// </summary>
+        [TestMethod]
+        public void ScriptCreateWithDbNullRangeCorrectly_PartitionFunction()
+        {
+            ExecuteFromDbPool(db =>
+            {
+                // Create a Partition Function using SMO
+                var partitionFunction = new _SMO.PartitionFunction(db, "PF_NULL_RANGE_TEST");
+                partitionFunction.RangeType = _SMO.RangeType.Left;
+                partitionFunction.PartitionFunctionParameters.Add(new _SMO.PartitionFunctionParameter(partitionFunction, _SMO.DataType.Int));
+                partitionFunction.RangeValues = new object[] { DBNull.Value, 10, 100 };
+                partitionFunction.Create();
+
+                var scripter = new _SMO.Scripter(db.Parent);
+                scripter.Options.ScriptForCreateOrAlter = true;
+                scripter.Options.EnforceScriptingOptions = true;
+                scripter.Options.ContinueScriptingOnError = true;
+                var scripts = scripter.EnumScript(partitionFunction).ToArray();
+                Assert.AreEqual(scripts.Length, 1);
```

Comment:
Add a comment that inform whoever is running the test and getting an error about what went wrong.

Also, I don't know about you, but I always get confused by which param is "expected" and which one is "actual", just like you here :-)

Two options:
1. Assert.That(scripts.Length, Is.EqualTo(1), $"Oh no!");
2. Assert.AreEqual(expected: 1, actual: Length, "What?!?");

David and I like the fluent assertions a lot (he does a lot more and he is bad influence on me... :)